packages/react-codemirror/src/CypherEditor.tsx

```diff
@@ -302,7 +302,13 @@ export class CypherEditor extends Component<CypherEditorProps> {
       });
     }
 
-    if (prevProps.history?.length !== this.props.history?.length) {
+    // This component rerenders on every keystroke and comparing the
+    // full lists of editor strings on every render could be expensive.
+    const didChangeHistoryEstimate =
+      prevProps.history?.length !== this.props.history?.length ||
```

Comment:
One question. Can this array grow indefinitely?

Reply:
It's up to the consumer, in workspace we currently limit it to 300. In browser it's configurable by the user. Some of the query strings can also be quite large